### DISCUSSION TITLE
XCode 16 Explicitly Built Modules not taking effect for SPM-packaged apps?

### DISCUSSION THREAD

**Discussion by beetee17** [pointfreeco/isowords#203] [Q&A]
Hi all,

XCode 16 introduced a new build system that seeks to improve build times and reliability. Following the [WWDC 24 talk Demystify Explicitly Built Modules](https://developer.apple.com/videos/play/wwdc2024/10171/), I installed XCode 16 Beta 1 and opted in to the new build system for my project by enabling the Explicitly Built Modules build setting for all targets (just to be sure). 

For reference, my app’s code is wrapped in an SPM package, very similar to the set-up used here. I was expecting that the new build system would explicitly build the modules within the package. However, when inspecting the build logs, it seems that the new system is not taking effect as, for example, there are no Scanning Dependency tasks. 

As a quick test, I cloned this repo and tried building with Explicitly Built Modules enabled, and observed similar results.

I am curious to know if this is the expected behaviour, i.e the new build system does not apply to modules in an SPM package? Otherwise, are there any steps I can take to get the new build system to take effect?

Thank you!

**Comment by mbrandonw** [comment]
Hi @beetee17, we haven't taken the time to look into the new build system or explicit modules yet, so we are not sure. This sounds like a question for Apple on their forums or possibly file a Feedback.

**Reply by beetee17** [reply]
Hi @mbrandonw, I’ve indeed made a similar post on the Apple Developer forums, though my experience on that site has led me to believe I’d probably have better luck elsewhere, hence my post here. 

My apologies for the wording that made it sound like I was pointing out an issue with isowords, my intention with the post was just to ask if anyone had a similar experience with the new build system, assuming that a good portion of the audience would have structured their apps in a similar fashion as isowords, or have experience with SPM in general. 

I appreciate you still taking the time to respond, and if you manage to try out the new XCode, I’d love to know!
